### PR TITLE
Add url method

### PR DESCRIPTION
**Pull request by msabramo**
Convert a path to a URL with 'file' scheme

POSIX:

    >>> path('/foo/bar/dog.txt').url()
    'file:///foo/bar/dog.txt'

Windows:

    >>> path('C:\Python27\README.txt').url()
    'file:///C:/Python27/README.txt'

### CODE REVIEW

**Comment by jaraco** [comment]
Does it handle relative paths?

**Comment by jaraco** [comment]
I see that the tests do cover relative paths. However, in my experience, the relative paths should have only two leading slashes.

**Comment by jaraco** [comment]
I'll have to dig around to find where I encountered this finding.

**Comment by msabramo** [comment]
I found this. 

https://github.com/pypa/pip/issues/328

I'll have to check later to see how compliant the code is with the RFC mentioned above.

**Comment by jaraco** [comment]
Thanks for that reference. It's actually more helpful than the code I was trying to reference.

Based on my reading, path.py should invoke os.path.isabs on the path. If it's absolute, use the implementation indicated. If it's not absolute, only prefix 'file:'. In all cases, substitute os.pathsep for posixpath.pathsep.

File: test_path.py [hunk 1]
```diff
@@ -937,5 +937,21 @@ def test_exception_in_context(self, tmpdir):
         assert not 'Lorem' in data
         assert 'lazy dog' in data
 
+
+@pytest.mark.skipif(sys.platform.startswith('win'),
```
Comment (jaraco) [inline test_path.py]: This test should pass on Windows as well. No need to skip it.

File: test_path.py [hunk 2]
```diff
@@ -937,5 +937,21 @@ def test_exception_in_context(self, tmpdir):
...
+def test_posix_url():
+    assert Path('/var/log/xyzzy').url() == 'file:///var/log/xyzzy'
+    assert Path('xyzzy').url() == 'file:///xyzzy'
+
+
+@pytest.mark.skipif(not sys.platform.startswith('win'),
+                    reason="requires Windows")
+def test_windows_url():
+    assert Path(r'Foo\Bar').url() == 'file:///Foo/Bar'
+    assert Path(r'C:/Foo/Bar').url() == 'file:///C://Foo/Bar'
```
Comment (jaraco) [inline test_path.py]: I don't understand the motivation for the extra slash here. That seems buggy to me. I'm tempted to punt on pathname2url. Although it does appear to be producing URLs that are openable by urlopen, it doesn't seem to be producing URLs that are consistent across platforms or conforming to the RFC. For example, "Foo\Bar" should resolve to something different than "\Foo\Bar".

File: test_path.py (hunk 1 above)
Comment (msabramo) [inline test_path.py]: Oh cool. I just removed the skip and it passes like you said it would. Thanks!

**Comment by jaraco** [comment]
I don't think this PR can be merged as-is. It's necessary that a relative path and absolute path be distinguishable. If stdlib modules can't do that properly, `path.py` should provide the proper implementation.

**Comment by jaraco** [comment]
I still would accept a patch for this functionality. Perhaps I was being too harsh in the critique. Perhaps it makes sense to accept the functionality and refine it, rather than get it perfect the first time.

**Comment by jaraco** [comment]
@msabramo If you would be willing to rebase your changes onto the latest master, I'll accept it as-is.

**Pull request closed without merging**